src/lib/api_utils.ts

```diff
@@ -61,9 +54,12 @@ async function fetchWebApi(
   let results: any[] = [];
   let fetchedSoFar: number = 0;
   let offset: number = 0;
-  let total: number;
+  let total: number = 0;
 
   do {
+    const apiUrl: URL = new URL(
```

Comment:
Moved this inside the do while loop so that the query params get reset each time and not continuously appended